File: frontend/console/src/api/signal-di.js
import Axios from "@/http";

export function createSignalDi(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-di', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveSignalDi(signalDiId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/signal-di/${ signalDiId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateSignalDi(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/signal-di', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalDi(signalDiId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/signal-di/${ signalDiId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalDiWithFlag(signalDiId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/signal-di/${ signalDiId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllSignalDis() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/signal-di/all').then(res => {
            resolve(res)
        })
    })
}

export function querySignalDi (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-di/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/signal_di.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_SIGNAL_DI = "/config/signal-di"
const URL_ROUTER_SIGNAL_DI_BY_ID = "/config/signal-di/:signal-di-id"
const URL_ROUTER_SIGNAL_DI_QUERY = "/config/signal-di/query"
const URL_ROUTER_SIGNAL_DI_ALL = "/config/signal-di/all"

func SignalDiRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_SIGNAL_DI, config.CreateSignalDi)
	r.GET(URL_ROUTER_SIGNAL_DI_BY_ID, config.RetrieveSignalDi)
	r.PUT(URL_ROUTER_SIGNAL_DI, config.UpdateSignalDi)
	r.DELETE(URL_ROUTER_SIGNAL_DI_BY_ID, config.DeleteSignalDi)
	r.POST(URL_ROUTER_SIGNAL_DI_QUERY, config.QuerySignalDi)
	r.GET(URL_ROUTER_SIGNAL_DI_ALL, config.GetAllSignalDis)
	return r
}